cordformation/src/main/kotlin/net/corda/plugins/Dockerform.kt

```diff
@@ -177,4 +182,24 @@ open class Dockerform @Inject constructor(objects: ObjectFactory) : Baseform(obj
 
         Files.write(dockerComposePath, dockerComposeContent.toByteArray())
     }
+
+    private fun encodeTypesafeURL(url: String): String {
+        val matcher = Pattern.compile("\\$\\{([^}]+)\\}").matcher(url);
+
+        val builder = StringBuilder()
+        var endOfPrevTokenIndex = 0
+
+        while(matcher.find()) {
+            val startIndex = matcher.start()
+            val token = matcher.group(1)
+            if (startIndex != 0) {
+                builder.append("\"${url.substring(endOfPrevTokenIndex, startIndex)}\"")
+            }
+            val encodedToken = "\${$token}"
+            builder.append(encodedToken)
+            endOfPrevTokenIndex = startIndex + encodedToken.length
+        }
+        builder.append(url.substring(endOfPrevTokenIndex))
+        return builder.toString()
+    }
```

Comment:
And the unit tests for this function...? :smiling_imp:
Don't worry, I don't mean full-blown Gradle plugin unit tests for this! Simple tests for this function alone will suffice. So a Kotlin equivalent of:
> When we encode "templateA" and pass it to Typesafe Config, the final string is "valueA" :heavy_check_mark: 
> When we encode "junkTemplateB" and pass it to Typesafe Config, we get "Error B" :skull_and_crossbones:

Reply:
Moved the encode function to a class and added unit tests 🙏